### PR TITLE
Fixed #25723 -- Made related field checks lookup against their local apps.

### PR DESCRIPTION
**Pull request by charettes**
Discovered while working on #5613

### CODE REVIEW

File: tests/invalid_models_tests/test_relative_fields.py [hunk 1]
```diff
@@ -132,6 +133,45 @@ class Model(models.Model):
...
+                apps = other_apps
+
+        class Model(models.Model):
+            foreign_key = models.ForeignKey(OtherModel, models.CASCADE)
+
+        field = Model._meta.get_field('foreign_key')
+        errors = field.check()
+        expected = [
+            Error(
+                ("Field defines a relation with model 'OtherModel', "
```
Comment (timgraham) [inline tests/invalid_models_tests/test_relative_fields.py]: I would omit the parenthesis in these messages (I know it's done elsewhere, but "I am at war" with that style unless you like it).

**Comment by timgraham** [comment]
I don't know that we need two tests for each modification -- did you have a thought about that? Some comments/docstrings in the tests might make them easier to browse outside the context of this pull request.

File: tests/invalid_models_tests/test_relative_fields.py (hunk 1 above)
Comment (charettes) [inline tests/invalid_models_tests/test_relative_fields.py]: I don't particularly like it, I'll change it. It was a copy-paste from the similar tests in this module.

**Comment by charettes** [comment]
There's two checks for each modification because I started by adding the *negative* ones since the test app is named `invalid_model_tests` (which should be named `model_checks` IMO) and then I realised it included many *positive* test already. I'd prefer to keep the *positive* version unless you think otherwise.

I'll also add comments/docstrings for future reference.

**Comment by timgraham** [comment]
Sounds good.

**Pull request merged**